contracts/teller-network/MetadataStore.sol

```diff
@@ -0,0 +1,212 @@
+        require(_margin <= 100, "Margin too high");
+
+        if (!sellerWhitelist[msg.sender]) {
+            Seller memory seller = Seller(_statusContractCode, _location, _username);
+            uint256 sellerId = sellers.push(seller) - 1;
+            addressToSeller[msg.sender] = sellerId;
+            sellerWhitelist[msg.sender] = true;
+        } else {
+            Seller storage tmpSeller = sellers[addressToSeller[msg.sender]];
+            tmpSeller.statusContractCode = _statusContractCode;
+            tmpSeller.location = _location;
+            tmpSeller.username = _username;
+        }
+        
+        Offer memory offer = Offer(_asset, _currency, _margin, _paymentMethods, _marketType, OfferStatus.Open);
+        uint256 offerId = offers.push(offer) - 1;
+        offerWhitelist[msg.sender][offerId] = true;
+        addressToOffers[msg.sender].push(offerId);
+
+        emit Added(
+            msg.sender, offerId, _asset, _statusContractCode, _location, _currency, _username, _paymentMethods, _marketType, _margin, OfferStatus.Open
+        );
+    }
+
+    /**
+    * @dev Update the seller
+    * @param _asset The address of the erc20 to exchange, pass 0x0 for Eth
+    * @param _statusContractCode The address of the status contract
+    * @param _location The location on earth
+    * @param _currency The currency the seller want to receive (USD, EUR...)
+    * @param _username The username of the seller
+    * @param _paymentMethods The list of the payment methods the seller accept
+    * @param _marketType Above or Below
+    * @param _margin The margin for the seller from 0 to 100
+    */
+    function update(
+        uint256 _offerId,
+        address _asset,
+        address _statusContractCode,
+        string memory _location,
+        string memory _currency,
+        string memory _username,
+        PaymenMethods[] memory _paymentMethods,
+        MarketType _marketType,
+        uint8 _margin
+    ) public {
+        require(sellerWhitelist[msg.sender], "Seller do not exists");
+        require(offerWhitelist[msg.sender][_offerId], "Offer do not exists");
+        require(_margin <= 100, "Margin too high");
+
+        Seller storage tmpSeller = sellers[addressToSeller[msg.sender]];
+        tmpSeller.statusContractCode = _statusContractCode;
+        tmpSeller.location = _location;
+        tmpSeller.username = _username;
+        
+        offers[_offerId].asset = _asset;
+        offers[_offerId].currency = _currency;
+        offers[_offerId].paymentMethods = _paymentMethods;
+        offers[_offerId].marketType = _marketType;
+        offers[_offerId].margin = _margin;
+
+        emit Updated(msg.sender, _offerId, _asset, _statusContractCode, _location, _currency, _username, _paymentMethods, _marketType, _margin);
+    }
+
+    /**
+    * @dev Rate a seller from 0 to 4
+    * @param _to The address to rate
+    * @param _value The rate value
+    */
+    function rate(address _to, RateValue _value) public {
```

Comment:
We already have a rating for escrows. Do we transfer it here? It's probably easier to implement if it's inside the Seller, but it might make less sense, because you can rate someone you never had an escrow with.